src/api/make.js

```diff
@@ -52,9 +52,29 @@ export default async (providedOptions = {}) => {
     forgeConfig = await getForgeConfig(dir);
   });
 
-  if (platform && platform !== process.platform && !(process.platform === 'darwin' && platform === 'mas')) {
-    throw "You cannot run 'make' for a platform other than your system's platform";
-  }
+  const makers = {};
+  const targets = overrideTargets || forgeConfig.make_targets[platform];
+
+  targets.forEach((target) => {
+    const maker = requireSearchRaw(__dirname, [
+      `../makers/${platform}/${target}.js`,
+      `../makers/generic/${target}.js`,
+      `electron-forge-maker-${target}`,
+      target,
+      path.resolve(dir, target),
+      path.resolve(dir, 'node_modules', target),
+    ]);
+
+    if (!maker) {
+      throw `Could not find a build target with the name: ${target} for the platform: ${platform}`;
+    }
+
+    if (!maker.supportsPlatform) {
```

Comment:
`supportsPlatform` feels more like a query (more function like) that a property, could we actually change it to be a function (maybe even async to allow users to check for things like binaries being installed)

```js
if (!await maker.isSupportedOnCurrentPlatform()) {
```

This would allow third party makers (or even our own) to check for required binaries in an async way before declaring support.